contrib/segyconverter/utils/segyextract.py

```diff
@@ -0,0 +1,332 @@
+    :param str output_dir: path to directory where npy files will be written/rewritten
+    :param str prefix: prefix to use when writing npy files
+    :param int iline: iline header byte location
+    :param int xline: crossline header byte location
+    :returns: 3 dimentional numpy array of SEGY data
+    :rtype: nparray
+    """
+    fast_distinct, slow_distinct, trace_headers, sampledepth = get_segy_metadata(input_file, iline, xline)
+    with segyio.open(input_file, ignore_geometry=True) as segy_file:
+        segy_file.mmap()
+
+        fast_line_space = abs(fast_distinct[1] - fast_distinct[0])
+
+        slow_line_space = abs(slow_distinct[0] - slow_distinct[1])
+        sample_size = len(segy_file.samples)
+        layer_fastmax = max(fast_distinct)
+        layer_fastmin = min(fast_distinct)
+        layer_slowmax = max(slow_distinct)
+        layer_slowmin = min(slow_distinct)
+        layer_trace_ids = trace_headers[(trace_headers.fast >= layer_fastmin)
+                                        & (trace_headers.fast <= layer_fastmax)
+                                        & (trace_headers.slow >= layer_slowmin)
+                                        & (trace_headers.slow <= layer_slowmax)]
+
+        block = np.zeros((len(fast_distinct), len(slow_distinct), sampledepth), dtype=np.float32)
+        for _, row in layer_trace_ids.iterrows():
+            block[(row[FAST] - layer_fastmin) // fast_line_space,
+                  (row[SLOW] - layer_slowmin) // slow_line_space,
+                  0:sample_size] = segy_file.trace[row.name]
+
+        np.save(os.path.join(output_dir, "{}_{}_{}_{:05d}".format(prefix, fast_distinct[0],
+                             slow_distinct[0], 0)), block)
+        variance = np.var(block)
+        stddev = np.sqrt(variance)
+        mean = np.mean(block)
+
+        with open(os.path.join(output_dir, prefix + '.json'), 'w') as f:
+            f.write(json.dumps({'stddev': str(stddev), 'mean': str(mean)}))
+        print("Npy files written: 1")
+    return block
+
+
+def process_segy_data(input_file, output_dir, prefix,
+                      iline=189, xline=193, n_points=128, stride=128):
+    """
+    Open segy file and write all numpy array files to disk
+    :param str input_file: path to segyfile
+    :param str output_dir: path to directory where npy files will be written/rewritten
+    :param str prefix: prefix to use when writing npy files
+    :param int iline: iline header byte location
+    :param int xline: crossline header byte location
+    :param int n_points: output cube size
+    :param int stride: stride when writing data
+    """
+    fast_indexes, slow_indexes, trace_headers, _ = get_segy_metadata(input_file, iline, xline)
+    with segyio.open(input_file, ignore_geometry=True) as segy_file:
+        segy_file.mmap()
+        # Global variance of segy data
+        variance = 0
+        mean = 0
+        sample_count = 0
+        filecount = 0
```

Comment:
config again

Reply:
These variables are not configurable. They are initializing counters to calculate the standard deviation if the file is too big to load into memory